master/internal/api_generic_tasks.go

```diff
@@ -279,6 +292,11 @@ func (a *apiServer) CreateGenericTask(
 		return nil, err
 	}
 
+	err = a.PostGenericTaskSpec(ctx, taskID, *genericTaskSpec, allocationID)
```

Comment:
```suggestion
	err = a.PersistGenericTaskSpec(ctx, taskID, *genericTaskSpec, allocationID)
```

I'd suggest using `Persist` name for database persistence functions over `Post` which triggers the associations with http POST.